test/supports/type_assertions.py

```diff
@@ -100,3 +102,23 @@ async def async_block() -> None:
 assert_type(file_io2.read(), bytes)
 assert_type(file_io2.readline(), bytes)
 assert_type(file_io2.readlines(), typing.Sequence[bytes])
+
+# check secrets
+secret = modal.Secret.from_dict({})
+assert_type(secret, modal.Secret)
+
+secret = modal.Secret.from_local_environ(["PATH"])
+assert_type(secret, modal.Secret)
+
+try:
```

Comment:
I think you don't have to do all of the "support" stuff here - the file should never be executed, it's only type checked